core/actions/operation.ts

```diff
@@ -75,6 +77,10 @@ export class Operation extends ActionBuilder<dataform.Operation> {
   // We delay contextification until the final compile step, so hold these here for now.
   private contextableQueries: Contextable<ICommonContext, string | string[]>;
 
+  // This is Action level flag. If set to true, we will add assertions from all the depenedncies of
+  // current action as dependencies. 
+  public dependOnDependencyAssertions: boolean = false;
```

Comment:
I don't see this class variable being read from anywhere - can't we remove it, and just use the value from the config?

Reply:
this variable is used inside `dependencies` on line 165. 
we do not attach config directly with the object. so when `dependencies` is called through ref() function, we need this information. thats why it is stored in this variable from config on an earlier stage.

Reply:
Ah I see what you mean, we can't store and read it from the core proto, like we do with most other fields, because it's not a field there. This makes sense then.

Reply:
In the future once we've migrated to just using the config proto, we can store the config proto and read this field from there